### DISCUSSION TITLE
Is WPF cross platform now

### DISCUSSION THREAD

**Discussion by mhdshameel** [dotnet/wpf#7275] [General]
For a development requirement, I came back to WPF after 4 years and find a lot of confusing changes and technological transitions. I used .NET Framework 3.5 earlier and now wish to pick on the latest technology available for WPF.

This [announcement](https://github.com/dotnet/wpf/issues/1936) announces the successful porting of WPF to the .NET core.
From this [article](https://www.softensity.com/blog/differences-between-net-core-3-1-and-net-core-6/), I came to know that .NET core is built primarily keeping cross-platform in mind. And I tried building a sample WPF project and Visual Studio recommends me to go with .NET Core 3.1 as shown below.

![image](https://user-images.githubusercontent.com/16662695/201006303-5d9a22dd-c6d3-4476-95e0-b06a30f8338d.png)

If .NET Core is cross-platform, then is WPF cross-platform now? Doesn't WPF abbreviates to **Windows** Presentation Foundation, what is the motivation behind porting it into a cross platform ready framework.

**Comment by miloush** [comment]
WPF is not cross-platform. .NET Core (now .NET) has platform specific targets and WPF is using those. You might need to get newer templates/SDK/VS, .NET 6 is the newest one with long-term support and .NET 7 was released a few days ago. See https://learn.microsoft.com/en-us/lifecycle/products/microsoft-net-and-net-core.

**Comment by lindexi** [comment]
How about MAUI ?

**Reply by saint4eva** [reply]
Maui is cross platform - it runs on Windows, Android, iOS, MacOS, Tizen

**Reply by greatawesome** [reply]
Missing Linux. This is a problem, because embedded Linux is the 2nd most important target for me outside of Windows.  It is also a target that all of the other .NET cross platform toolkits support.  MAUI is the outlier.

**Reply by saint4eva** [reply]
when the demand for Linux really increases, the team will tackle it.

**Comment by greatawesome** [comment]
There is an option to build cross platform WPF applications now: https://avaloniaui.net/XPF

It requires a commercial license to use.

**Reply by saint4eva** [reply]
when other options are open source?
